File: TypeScript/src/count-char.ts
function countChar(tmpString) {
    let counts = new Map();
    for (let letter of tmpString) {
        if (counts.has(letter)) {
            counts.set(letter, counts.get(letter) + 1)
        } else {
            counts.set(letter, 1)
        }
    }

    const countsSorted = new Map([...counts.entries()].sort());
    countsSorted.forEach((value: number, key: string) => {
        let percent = value / tmpString.length;
        console.log(`'${key}' ${value} ${percent.toFixed(2)}`);
    });
}

let inputString = "12345678900000000000-Hello-World";
countChar(inputString);

File: Rust/src/count_char.rs
use std::collections::BTreeMap;

fn main() {
    let input_string: &str = "12345678900000000000-Hello-World";
    count_char(input_string);
}

fn count_char(tmp_string: &str) {
    let mut counts: BTreeMap<char, u32> = BTreeMap::new();

    for letter in tmp_string.chars() {
        let counter: &mut u32 = counts.entry(letter).or_insert(0);
        *counter += 1;
    }

    for (letter, value) in counts.iter() {
        let percent = *value as f32 / tmp_string.len() as f32 as f32;
        println!("'{}' {} {:.2}", letter, value, percent);
    }
}
